.changelog/2887.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/mongodbatlas_search_index: Fixes resource create and update when `wait_for_index_build_completion` attribute is used
```

Comment:
Can we add a test making use of `wait_for_index_build_completion` attribute?

Reply:
thanks, done in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2887/commits/5b0fbe35ceeefa2a64c5ec063118c89dadac4316

Reply:
hi @AgustinBettati , I'm reverting the test because it's failing with: `HTTP 404 Not Found`
I think it's because when the database and collection doesn't exist the index eventually fails. In my manual tests where I create db and collection, it works fine